src/View/mainpage/IsodoseTab.py

```diff
@@ -67,65 +70,26 @@ def init_checkboxes(self):
+        list_of_doses = []
+        for percentage in isodose_percentages:
+            dose = int(self.rx_dose_in_cgray * (percentage / 100))
+            list_of_doses.append(dose)
 
         # Checkboxes
-        checkbox1 = QtWidgets.QCheckBox("107 % / " + str(val1) + " cGy [Max]")
-        checkbox2 = QtWidgets.QCheckBox("105 % / " + str(val2) + " cGy")
-        checkbox3 = QtWidgets.QCheckBox("100 % / " + str(val3) + " cGy")
-        checkbox4 = QtWidgets.QCheckBox("95 % / " + str(val4) + " cGy")
-        checkbox5 = QtWidgets.QCheckBox("90 % / " + str(val5) + " cGy")
-        checkbox6 = QtWidgets.QCheckBox("80 % / " + str(val6) + " cGy")
-        checkbox7 = QtWidgets.QCheckBox("70 % / " + str(val7) + " cGy")
-        checkbox8 = QtWidgets.QCheckBox("60 % / " + str(val8) + " cGy")
-        checkbox9 = QtWidgets.QCheckBox("30 % / " + str(val9) + " cGy")
-        checkbox10 = QtWidgets.QCheckBox("10 % / " + str(val10) + " cGy")
-
-        checkbox1.clicked.connect(lambda state, text=107: self.checked_dose(state, text))
-        checkbox2.clicked.connect(lambda state, text=105: self.checked_dose(state, text))
-        checkbox3.clicked.connect(lambda state, text=100: self.checked_dose(state, text))
-        checkbox4.clicked.connect(lambda state, text=95: self.checked_dose(state, text))
-        checkbox5.clicked.connect(lambda state, text=90: self.checked_dose(state, text))
-        checkbox6.clicked.connect(lambda state, text=80: self.checked_dose(state, text))
-        checkbox7.clicked.connect(lambda state, text=70: self.checked_dose(state, text))
-        checkbox8.clicked.connect(lambda state, text=60: self.checked_dose(state, text))
-        checkbox9.clicked.connect(lambda state, text=30: self.checked_dose(state, text))
-        checkbox10.clicked.connect(lambda state, text=10: self.checked_dose(state, text))
-
-        checkbox1.setStyleSheet("font: 10pt \"Laksaman\";")
-        checkbox2.setStyleSheet("font: 10pt \"Laksaman\";")
-        checkbox3.setStyleSheet("font: 10pt \"Laksaman\";")
-        checkbox4.setStyleSheet("font: 10pt \"Laksaman\";")
-        checkbox5.setStyleSheet("font: 10pt \"Laksaman\";")
-        checkbox6.setStyleSheet("font: 10pt \"Laksaman\";")
-        checkbox7.setStyleSheet("font: 10pt \"Laksaman\";")
-        checkbox8.setStyleSheet("font: 10pt \"Laksaman\";")
-        checkbox9.setStyleSheet("font: 10pt \"Laksaman\";")
-        checkbox10.setStyleSheet("font: 10pt \"Laksaman\";")
-
-        list_of_checkboxes.append(checkbox1)
-        list_of_checkboxes.append(checkbox2)
-        list_of_checkboxes.append(checkbox3)
-        list_of_checkboxes.append(checkbox4)
-        list_of_checkboxes.append(checkbox5)
-        list_of_checkboxes.append(checkbox6)
-        list_of_checkboxes.append(checkbox7)
-        list_of_checkboxes.append(checkbox8)
-        list_of_checkboxes.append(checkbox9)
-        list_of_checkboxes.append(checkbox10)
+        first_iteration = True
+        for i in range(10):
```

Comment:
or have one line for i = 0 and then range(1:10) for the rest.

while the following is not ideal for this situation, I've learned about enumerate
```
for i, isodose_percent in enumerate(isodose_percentages):
  print(i, isodose_percent) # or whatever else you need the count/index and value for...
```

Reply:
That's an interesting and powerful function, thanks for bringing it to my attention.